src/libs/SearchUtils.ts

```diff
@@ -534,10 +536,10 @@ function buildQueryStringFromFilterValues(filterValues: Partial<SearchAdvancedFi
             return `${value}`;
         }
         if (filterKey === FILTER_KEYS.TYPE && filterValue) {
-            return `${CONST.SEARCH.SYNTAX_ROOT_KEYS.TYPE}:${sanitizeString(filterValue as string)}`;
+            expenseFilter = `${CONST.SEARCH.SYNTAX_ROOT_KEYS.TYPE}:${sanitizeString(filterValue as string)}`;
         }
         if (filterKey === FILTER_KEYS.STATUS && filterValue) {
-            return `${CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS}:${sanitizeString(filterValue as string)}`;
+            statusFilter = `${CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS}:${sanitizeString(filterValue as string)}`;
```

Comment:
It is a bit dirty that we are inside a `.map()` where you are supposed to always return a value, and suddenly in just 2 cases we are instead writing into a variable outside of the loop and then using this value.
Since we're in a map won't this produce `undefined` values inside the resulting `filterString` array?
and then if you run a `join(',')` over an array with undefined values this you will get dangling commas?

IMO we should rewrite this code to work differently, for example first destructure `filterValues` object in the first line, to grab `type` and `status` filter, handle them, then loop over the rest

Reply:
I didn't want to introduce potential side effects to a method I didn't have prior context on, but I do agree with what you've said it'll be cleaner.

I've updated code.